### PR TITLE
Sct 854

### PR DESCRIPTION
**Pull request by sean-mccorkle**
This is a fix to the provenance of the filtered expression matrix  - it now has a proper link to the differential expression matrix used for filtering.

### CODE REVIEW

File: test/FeatureSetUtils_server_test.py [hunk 1]
```diff
@@ -307,6 +307,32 @@ def test_upload_featureset_from_diff_expr(self):
...
+
+        # check each filtered expression matrix in the set:
+
+        for fem in result.get('filtered_expression_matrix_ref_list'):
+            prov = self.wsClient.get_object_provenance([{'ref': fem}])[0].get('provenance')[0]
+            self.assertTrue(prov.get( 'input_ws_objects'))
+            dem_list = prov.get('input_ws_objects')
+            self.assertTrue(isinstance(dem_list, list))
+            self.assertTrue(len( dem_list ) == 1)    # should be a list of one
+            self.assertTrue(dem_list[0] in dms)      # and in the diff. expr. set list
```
Comment (JamesJeffryes) [inline test/FeatureSetUtils_server_test.py]: Each of the dem_list should be unique right? Maybe check this to make sure you don't get duplicates?

Comment (sean-mccorkle) [reply test/FeatureSetUtils_server_test.py]: Good idea.  Unfortunately the unit test has duplicates!   (I didn't want to spend time fixing that)

Comment (sean-mccorkle) [reply test/FeatureSetUtils_server_test.py]: Do you think its worth the time and effort?  maybe..

Comment (JamesJeffryes) [reply test/FeatureSetUtils_server_test.py]: Up to you. Mostly want to catch a case where you are returning only one ref but if it's more than one line of code feel free to ignore

Comment (sean-mccorkle) [reply test/FeatureSetUtils_server_test.py]: Actually now that I look again - dem_list should always be a list of exactly one.   Maybe the variable name is misleading

Comment (JamesJeffryes) [reply test/FeatureSetUtils_server_test.py]: Fair enough

**Pull request merged**